src/model/table.py

```diff
@@ -48,51 +70,40 @@ def __get_efficient_index__(self, query):
 
     def __get_all_primary_keys__(self):
         primary_keys = []
-        for primary_key_path in os.listdir(self.__get_data_path__()):
-            primary_key = self.__get_primary_key_from_path__(primary_key_path)
-            primary_keys.append(primary_key)
+        for primary_key in os.listdir(self.get_data_path()):
+            if primary_key != "Lock":
+                primary_keys.append(primary_key.replace(".json",""))
         return primary_keys
 
     def get_by_primary_key(self, primary_key):
-        path = self.__get_primary_key_path__(primary_key)
-        if not os.path.exists(path):
-            return None
-        with open(path, 'r') as file:
-            return json.load(file)
+        return Row.load_by_primary_key(self, primary_key)
 
     @staticmethod
     def __filter_by_query__(found_objects, query):
         if not query or str(query).isspace():
             return found_objects
         filtered_objects = []
         for object_to_compare in found_objects:
-            if Table.compare(object_to_compare, query):
+            if object_to_compare and object_to_compare.compare(query):
```

Comment:
Method name should be compare, should be has_attributes because u are not comparing